src/rust-util/pelikan-sys/src/lib.rs

```diff
@@ -206,12 +216,12 @@ pub mod core {
         }
     }
 
-    impl admin_options_st {
-        pub const fn new() -> Self {
+    unsafe impl ccommon::option::Options for admin_options_st {
+        fn new() -> Self {
             init_option! {
                 Self;
                 ACTION( admin_host,     OPTION_TYPE_STR,    std::ptr::null_mut(),       "admin interfaces listening on"),
-                ACTION( admin_port,     OPTION_TYPE_STR,    slice_to_ptr(ADMIN_PORT),   "admin port"                   ),
+                ACTION( admin_port,     OPTION_TYPE_STR,    crate::slice_to_ptr(ADMIN_PORT),   "admin port"                   ),
```

Comment:
Consider changing the formatting here, vertical alignment is thrown off my the change here. Might be better to make each action span multiple lines instead.

Reply:
I've fixed the alignment for this. For now I'd like to leave the formatting as-is so that it's as close to the way these are defined in ccommon as possible.